src/pyodide.js

```diff
@@ -11,20 +18,53 @@ var languagePluginLoader = new Promise((resolve, reject) => {
   ////////////////////////////////////////////////////////////
   // Package loading
   var packages = undefined;
-  let loadedPackages = new Set();
+  let loadedPackages = new Array();
+
+  let _uri_to_package_name = (package_uri) => {
+    // Generate a unique package name from URI
+
+    if (package_name_regexp.test(package_uri)) {
+      return package_uri;
+    } else if (package_uri_regexp.test(package_uri)) {
+      let match = package_uri_regexp.exec(package_uri);
+      // Get the regexp group corresponding to the package name
+      return match[1];
+    } else {
+      return null;
+    }
+  };
+
 
   let loadPackage = (names) => {
     // DFS to find all dependencies of the requested packages
     let packages = window.pyodide.packages.dependencies;
     let queue = new Array(names);
-    let toLoad = new Set();
+    let toLoad = new Array();
     while (queue.length) {
-      const package = queue.pop();
-      if (!loadedPackages.has(package)) {
-        toLoad.add(package);
+      let package_uri = queue.pop();
+
+      const package = _uri_to_package_name(package_uri);
+
+      if (package == null) {
+        console.log(`Invalid package name or URI '${package_uri}'`);
```

Comment:
So currently there is no mechanism to raise an error in iodide short of printing something to `console.log`?

Reply:
You can also `throw new Error(...)` if it really is fatal.  That will surface up to the Iodide interface, or outside of Iodide will end up in the console.